### PR TITLE
[BugFix] Fix insert into files with wrong format

### LINKED ISSUE #37806: [Bug] insert into files unable to use uppercase character PARQUET
### Steps to reproduce the behavior (Required)
1. execute sql
```sql
INSERT INTO 
FILES(
    'path' = 'hdfs://cluster/starrocks/export/',
    'format' = 'PARQUET',
    'hadoop.security.authentication' = 'simple',
    'compression' = 'lz4'
)

select * from table;
```

### Expected behavior (Required)
We just need check properties in fe

### Real behavior (Required)
There is no problem with fe node verification, but there is a problem with be node verification. 

fe log
```log
unsupported file formatPARQUET backend [id=10125] [host=starrocks-be-2.starrocks-be-search.bigdata.svc.cluster.local]
```

be log
```log
W1226 15:57:25.669840   745 internal_service.cpp:307] exec plan fragment failed, errmsg=unsupported file formatPARQUET
```

### StarRocks version (Required)
 - 3.2.1-79ee91d

### PR DESCRIPTION
**Pull request by Jin-H**
Why I'm doing:

What I'm doing:

Fixes https://github.com/StarRocks/starrocks/issues/37806

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by Jin-H** [comment]
@letian-jiang Can you help review it? The failure of sonar check is an unrelated problem. It should be that the sonar side has changed the jdk version rules.

**Comment by wanpengfei-git** [comment]
PR was manually closed.

**Comment by Jin-H** [comment]
why close this pr? @wanpengfei-git

**Comment by Jin-H** [comment]
I probably know why you close this PR. This PR is essentially to solve the problem that when  using `insert into files` statement, the format in files is filled in incorrectly (for example, orc is written), then the error report will say that the path needs to be modified, but this should actually be Modify the format field. The msg in the be code is just changed by the way.

for example

```sql
INSERT INTO 
FILES(
    'path' = 'hdfs://cluster/starrocks/export/',
    'format' = 'orc',
    'hadoop.security.authentication' = 'simple',
    'compression' = 'lz4'
)

select * from table;

```

the error msg will be `Use \"path\" = \"parquet\" as only parquet format is supported now`.

so this pr should reopen, thanks @wanpengfei-git

**Comment by letian-jiang** [comment]
Hi @Jin-H, it seems there are two points
1. `format` property should be case insensitive.
2. And there is a typo in the error message. `path` -> `format`

Am I right?

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/InsertStmt.java [hunk 1]
```diff
@@ -345,11 +347,11 @@ public Table makeTableFunctionTable(SessionVariable sessionVariable) {
 
         if (format == null) {
             throw new SemanticException("format is a mandatory property. " +
-                    "Use \"path\" = \"parquet\" as only parquet format is supported now");
+                    "Use \"format\" = \"parquet\" as only parquet format is supported now");
         }
 
-        if (!format.equalsIgnoreCase("parquet")) {
-            throw new SemanticException("use \"path\" = \"parquet\", as only parquet format is supported now");
+        if (!PARQUET_FORMAT.equals(format)) {
```
Comment (letian-jiang) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/InsertStmt.java:353]: why use `equals`?

**Comment by letian-jiang** [comment]
Thanks for your contribution! I think we only need to keep the relevant part of codes.

File: be/src/exec/pipeline/sink/table_function_table_sink_operator.cpp [hunk 2]
```diff
@@ -224,7 +224,7 @@ Status TableFunctionTableSinkOperatorFactory::prepare(RuntimeState* state) {
         }
         _parquet_file_schema = result.ValueOrDie();
     } else {
-        return Status::InternalError("unsupported file format" + _file_format);
+        return Status::InternalError("unsupported file format: " + _file_format);
```
Comment (letian-jiang) [inline be/src/exec/pipeline/sink/table_function_table_sink_operator.cpp:227]: I got you. The check at BE is case-sensitive. Can you modify the 219 line to make it case-insensitive?

Comment (letian-jiang) [reply be/src/exec/pipeline/sink/table_function_table_sink_operator.cpp:227]: maybe we can use `boost::iequals`

**Comment by Jin-H** [comment]
> 2\. And there is a typo in the error message. `path` -> `format`

yes

File: be/src/exec/pipeline/sink/table_function_table_sink_operator.cpp (hunk 2 above)
Comment (Jin-H) [reply be/src/exec/pipeline/sink/table_function_table_sink_operator.cpp:227]: ok

**Review by letian-jiang — APPROVED** [review]
LGTM

**Review by packy92 — APPROVED** [review]

**Comment by Jin-H** [comment]
@letian-jiang Please help me re-approve it, thanks

**Review by letian-jiang — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**